cmd/juju/cloud/show_test.go

```diff
@@ -86,26 +118,36 @@ clouds:
     regions:
       london:
         endpoint: http://london/1.0
+    config:
+      bootstrap-retry-delay: 1500
+      network: nameme
     region-config:
       london:
         bootstrap-timeout: 1800
+        use-floating-ip: true
 `[1:]
 	err := ioutil.WriteFile(osenv.JujuXDGDataHomePath("clouds.yaml"), []byte(data), 0600)
 
 	ctx, err := cmdtesting.RunCommand(c, cloud.NewShowCloudCommand(), "homestack")
 	c.Assert(err, jc.ErrorIsNil)
 	out := cmdtesting.Stdout(ctx)
-	c.Assert(out, gc.Equals, `
-defined: local
+	c.Assert(out, gc.Equals, strings.Join([]string{`defined: local
 type: openstack
 description: Openstack Cloud
 auth-types: [userpass, access-key]
 endpoint: http://homestack
 regions:
   london:
     endpoint: http://london/1.0
+bootstrap-config:
+  bootstrap-retry-delay: 1500
+model-config:
+  network: nameme
 region-config:
   london:
-    bootstrap-timeout: 1800
-`[1:])
+    bootstrap-config:
+      bootstrap-timeout: 1800
+    model-config:
+      use-floating-ip: true
+`, openstackProviderConfig}, ""))
```

Comment:
I'm starting to think that this block of provider specific config should be printed separately as it is not "what's defined in clouds.yaml" (which is what show-cloud nominally prints), but metadata. So something like below (ignore bad formatting). Sorry for churn, it's taken a few iterations to get a feel for it.

`$ juju show-cloud myopenstack
defined: local
type: openstack
description: Openstack Cloud
auth-types: [userpass, access-key]
endpoint: http://homestack
regions:
  london:
    endpoint: http://london/1.0
config:
  max-logs-age: 2m
  max-logs-size: 1M
region-config:
  london:
    max-logs-age: 4m

The available config options specific to openstack clouds are:
  external-network:
    type: string
    description: The network label or UUID to create floating IP addresses on when
      multiple external networks exist.
  network:
    type: string
    description: The network label or UUID to bring machines up on when multiple networks
        exist.
  use-default-secgroup:
    type: bool
    description: Whether new machine instances should have the "default" Openstack
      security group assigned in addition to juju defined security groups.
  use-floating-ip:
    type: bool
    description: Whether a floating IP address is required to give the nodes a public
      IP address. Some installations assign public IP addresses by default without
      requiring a floating IP address.`